src/client/keycode.cpp

```diff
@@ -29,219 +29,132 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 	wchar_t Char; // L'\0' means no character assigned
 	const char *LangName; // NULL means it doesn't have a human description
 };
 
-#define DEFINEKEY1(x, lang) /* Irrlicht key without character */ \
+#define DEFINEKEY1(x, lang) /* Irrlicht key without character */                         \
 	{ #x, irr::x, L'\0', lang },
-#define DEFINEKEY2(x, ch, lang) /* Irrlicht key with character */ \
+#define DEFINEKEY2(x, ch, lang) /* Irrlicht key with character */                        \
 	{ #x, irr::x, ch, lang },
-#define DEFINEKEY3(ch) /* single Irrlicht key (e.g. KEY_KEY_X) */ \
-	{ "KEY_KEY_" TOSTRING(ch), irr::KEY_KEY_ ## ch, (wchar_t) *TOSTRING(ch), TOSTRING(ch) },
-#define DEFINEKEY4(ch) /* single Irrlicht function key (e.g. KEY_F3) */ \
-	{ "KEY_F" TOSTRING(ch), irr::KEY_F ## ch, L'\0', "F" TOSTRING(ch) },
-#define DEFINEKEY5(ch) /* key without Irrlicht keycode */ \
-	{ ch, irr::KEY_KEY_CODES_COUNT, (wchar_t) *ch, ch },
+#define DEFINEKEY3(ch) /* single Irrlicht key (e.g. KEY_KEY_X) */                        \
+	{ "KEY_KEY_" TOSTRING(ch), irr::KEY_KEY_##ch, (wchar_t)*TOSTRING(ch), TOSTRING(ch) },
+#define DEFINEKEY4(ch) /* single Irrlicht function key (e.g. KEY_F3) */                  \
+	{ "KEY_F" TOSTRING(ch), irr::KEY_F##ch, L'\0', "F" TOSTRING(ch) },
+#define DEFINEKEY5(ch) /* key without Irrlicht keycode */                                \
+	{ ch, irr::KEY_KEY_CODES_COUNT, (wchar_t)*ch, ch },
 
 #define N_(text) text
 
 static const struct table_key table[] = {
 	// Keys that can be reliably mapped between Char and Key
-	DEFINEKEY3(0)
-	DEFINEKEY3(1)
-	DEFINEKEY3(2)
-	DEFINEKEY3(3)
-	DEFINEKEY3(4)
-	DEFINEKEY3(5)
-	DEFINEKEY3(6)
-	DEFINEKEY3(7)
-	DEFINEKEY3(8)
-	DEFINEKEY3(9)
-	DEFINEKEY3(A)
-	DEFINEKEY3(B)
-	DEFINEKEY3(C)
-	DEFINEKEY3(D)
-	DEFINEKEY3(E)
-	DEFINEKEY3(F)
-	DEFINEKEY3(G)
-	DEFINEKEY3(H)
-	DEFINEKEY3(I)
-	DEFINEKEY3(J)
-	DEFINEKEY3(K)
-	DEFINEKEY3(L)
-	DEFINEKEY3(M)
-	DEFINEKEY3(N)
-	DEFINEKEY3(O)
-	DEFINEKEY3(P)
-	DEFINEKEY3(Q)
-	DEFINEKEY3(R)
-	DEFINEKEY3(S)
-	DEFINEKEY3(T)
-	DEFINEKEY3(U)
-	DEFINEKEY3(V)
-	DEFINEKEY3(W)
-	DEFINEKEY3(X)
-	DEFINEKEY3(Y)
-	DEFINEKEY3(Z)
-	DEFINEKEY2(KEY_PLUS, L'+', "+")
-	DEFINEKEY2(KEY_COMMA, L',', ",")
-	DEFINEKEY2(KEY_MINUS, L'-', "-")
-	DEFINEKEY2(KEY_PERIOD, L'.', ".")
+	DEFINEKEY3(0) DEFINEKEY3(1) DEFINEKEY3(2) DEFINEKEY3(3) DEFINEKEY3(4) DEFINEKEY3(
+			5) DEFINEKEY3(6) DEFINEKEY3(7) DEFINEKEY3(8) DEFINEKEY3(9) DEFINEKEY3(A)
+			DEFINEKEY3(B) DEFINEKEY3(C) DEFINEKEY3(D) DEFINEKEY3(E) DEFINEKEY3(
+					F) DEFINEKEY3(G) DEFINEKEY3(H) DEFINEKEY3(I) DEFINEKEY3(J)
+					DEFINEKEY3(K) DEFINEKEY3(L) DEFINEKEY3(M) DEFINEKEY3(N) DEFINEKEY3(
+							O) DEFINEKEY3(P) DEFINEKEY3(Q) DEFINEKEY3(R) DEFINEKEY3(S)
+							DEFINEKEY3(T) DEFINEKEY3(U) DEFINEKEY3(V) DEFINEKEY3(W)
+									DEFINEKEY3(X) DEFINEKEY3(Y) DEFINEKEY3(Z) DEFINEKEY2(
+											KEY_PLUS, L'+', "+") DEFINEKEY2(KEY_COMMA,
+											L',', ",") DEFINEKEY2(KEY_MINUS, L'-', "-")
+											DEFINEKEY2(KEY_PERIOD, L'.', ".")
```

Comment:
![1376940511321](https://user-images.githubusercontent.com/1042418/84498292-aaeb7280-acb0-11ea-8469-a8571fc995bf.jpg)

Reply:
I suspect that's happening because the macro is written in a questionable sort of way, with the comma that's supposed to separate array entries being baked in instead of being there between each item.

Might be worth testing to see what happens if the commas are there, I would expect clang-format to handle this code properly then.